src/engines/nwn2/roster.cpp

```diff
@@ -0,0 +1,64 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  A roster array in a Neverwinter Nights 2 module.
+ */
+
+#include "src/aurora/types.h"
+#include "src/aurora/resman.h"
+#include "src/aurora/gff3file.h"
+
+#include "src/engines/nwn2/roster.h"
+
+namespace Engines {
+
+namespace NWN2 {
+
+Roster::Roster() {
+	load();
+}
+
+Roster::~Roster() {
+}
+
+/** Load roster from the saved 'ROSTER.rst' file, if any. */
+void Roster::load() {
+	Common::ScopedPtr<Common::SeekableReadStream> stream(ResMan.getResource("ROSTER", Aurora::kFileTypeRST));
+	if (!stream)
+		return;
+
+	Common::ScopedPtr<Aurora::GFF3File> gff(new Aurora::GFF3File(stream.release(), MKTAG('R', 'S', 'T', ' ')));
```

Comment:
Technically, `Aurora::GFF3File` already has a constructor that takes a resource name and type, and queries the ResourceManager.

It throws when it doesn't find the resource, so you'd need to catch the exception. Or, use `loadOptionalGFF3()` in `src/engines/aurora/util.h`, which already does that for you (it catches and ignores the exception, return 0 if the file can't be found).

I.e. just call

```
Common::ScopedPtr<Aurora::GFF3File> gff(loadOptionalGFF3("ROSTER", Aurora::kFileTypeRST, MKTAG('R', 'S', 'T', ' ')));
if (!gff)
	return;
```